### ISSUE TITLE
How to decide on additional themes?

### ISSUE THREAD

**Issue by tarnung** [200ok-ch/organice#539]
Now that we have the setup for it, it's rather easy to add additional themes.
I demonstrated that by researching some popular themes and adopting them for organice. (See  #487 for dark mode examples, see below for light mode examples.)
But how do we decide on themes that get included? Is there an upper limit? Should there be a balance between light and dark modes?

I'm ok with having many themes. But then the question becomes how they are chosen in the menu. Maybe a dropdown or a collapsing list?

What should I do with the themes I made so far? I could do individual draft-prs that could be promoted to live prs once someone shows interest in a certain theme. Or I could make one big pr with all the themes I consider ready in case we go with a 'more is better' approach.

gruvbox light
![gruvboxlight](https://user-images.githubusercontent.com/35081013/97202465-f14ef300-17b3-11eb-8d52-cef2b75e2c9e.png)

one light
![one-light](https://user-images.githubusercontent.com/35081013/97202487-fca21e80-17b3-11eb-886c-d8f76f15c2f0.png)

smyck light (there isn't supposed to be a light version, so consider this experimental)
![smyck-light](https://user-images.githubusercontent.com/35081013/97202516-0592f000-17b4-11eb-8b93-4d0b6c3192fe.png)

**Comment by munen** [comment]
It's a good question on how to decide. Here's my take:

Emacs is a very conservative piece of software. It's not easy to add new stuff and new adopters often find it quite ugly by default. So, they either don't use Emacs or go to a 'starter kit' like Doom or Spacemacs. Having said that, even Emacs itself ships with 14 themes[1] by default! So, even the conservative Emacs community deems having a 'lot' of themes by default something worthwhile.

Personally, I'm in favor that organice adopts different themes - now that you've put the necessary infrastructure in place to easily and semantically add new ones. So, even if we change the design/UX, the themes should kind of auto-update. Before, that would have been my biggest concern, it would have been easy to have regressions on design/UX changes. But now that all CSS is in one place, I think it's managable.

As for which themes to add, I'd apply this general rule of thumb: If it's something that a 'reasonably big contributor' of organice wants to have in organice, whilst it does not conflict with what already is built-in or the plans we have for the future, we can add it. By now, I'd easily put you into that 'reasonably big contributor' category. Also, you've made it so that adding a new theme is rather safe. So, I'd say: Whatever themes you want to add, I'm very happy to merge the PRs. My only other input would be: Don't put too much effort into what other people might want. Those kinds of people always want more and rarely give back. There's too much to do in the world to make them all happy(;

As for the interface, that's a good question. If all the themes that we include will have a light and dark mode, then we could have a 'theme' dropdown and a 'mode' dropdown. That would also still enable to listen to OS and browser defaults, because a switch is always possible. But then we'd stuck to only add themes that have both options. I'd be fine with that as long as you're currently not planning to add a theme that only has one option (even if it's a personal take and 'experimental' as you've noted above). If you plan to add just one that doesn't have this option, we'd need something else. I'm open for a UX suggestion there.

As for which to include when, I honestly pretty much like all the ones you've shared so far. I wouldn't use them all as a default, because I'm a picky person, and there can only be one default(; But, I can generally appreciate the looks of all of them (which is not to say that I always like themes!), so I assume that they might be the right default for somebody. Since they already look good, I'm happy if you want to include them all right now. That's also fine to have in one PR. I would just make sure to do testing of all views for all the themes, but that can be done quite quickly if there's no issues(;

NB: Emacs being conservative is not something that I personally consider negative(;

1. Emacs default themes 
- adwaita 
- deeper-blue 
- dichromacy 
- leuven 
- light-blue 
- manoj-dark 
- misterioso
- tango 
- tango-dark 
- tsdh-dark 
- tsdh-light 
- wheatgrass 
- whiteboard 
- wombat

**Comment by munen** [comment]
Addendum: It's quite amazing how many great themes you were able to port so quickly! I'm stoked that:

- This is possible now thanks to your great efforts (and a bit of my preparation by adding solarize and cleaning up beforehand a bit^^)
- You've put in the effort for that! :100: 

:pray: :bow:

**Comment by tarnung** [comment]
> If all the themes that we include will have a light and dark mode, then we could have a 'theme' dropdown and a 'mode' dropdown. That would also still enable to listen to OS and browser defaults.

I thought of that too. It's why I built that 'smyck light' (i simply inverted the base colors).
I'm by no means a themes person. So I can't say whether most themes have both a dark and light mode. I just saw your comment about it being easy to add a dark mode since solarized is built with that in mind and I thought 'let's see about that'. (You were right.) Then I enjoyed myself enough to keep going.

I'm actually rather happy with solarized light and I'd definitively keep it as a default for organice. But like @Naheel-Azawy I'd prefer a different theme for dark mode.

So I'll prepare a big bundle with pairs of light/dark themes.
